<h1 style="color:#38BDF8; text-align:center; font-weight:bold; border-bottom:3px solid #38BDF8;">Day 1 — Descriptive Statistics</h1>
<h3 style="text-align:center; color:#BE185D;">Central Tendency, Spread, Percentiles &amp; Quartiles</h3>


<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">Table of Contents</h2>

<div style="border:1px solid #ccc; padding:15px 25px; border-radius:6px;">
<ol style="font-weight:bold; line-height:1.9;">
<li><a href="#section0">0. Setup — Importing NumPy &amp; Pandas</a></li>
<li><a href="#section1">1. Why Statistics Comes Before Modeling</a></li>
<li><a href="#section2">2. Measures of Central Tendency — Mean, Median, Mode</a></li>
<li><a href="#section3">3. How Outliers Affect the Mean vs. the Median</a></li>
<li><a href="#section4">4. Measures of Spread — Range, Variance, Standard Deviation</a></li>
<li><a href="#section5">5. Percentiles, Quartiles &amp; the IQR</a></li>
<li><a href="#section6">6. Computing Everything at Once with Pandas <code>.describe()</code></a></li>
<li><a href="#section7">7. Common Mistakes to Avoid</a></li>
<li><a href="#section8">8. Quick Reference — Cheat Sheet</a></li>
<li><a href="#section9">9. Hands-On Lab — Describing a Real Dataset</a></li>
<li><a href="#section11">10. Summary — What I Learned Today</a></li>
</ol>
</div>


<a id="section0"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">0. Setup — Importing NumPy &amp; Pandas</h2>

<div style="border-left:5px solid #38BDF8; background-color:#eaf6ff; padding:10px 15px; margin:10px 0; border-radius:4px; color:#0c4a6e;">
<b>Note:</b> Descriptive statistics is really just <b>math applied to data</b>, so today's toolbox is small and familiar from Week 1: <b>NumPy</b> for the raw calculations, and <b>Pandas</b> for applying them to real tabular data with a single line of code.
</div>


In [1]:
import numpy as np
import pandas as pd

print("NumPy version:", np.__version__)
print("Pandas version:", pd.__version__)

NumPy version: 2.4.6
Pandas version: 3.0.3


<a id="section1"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">1. Why Statistics Comes Before Modeling</h2>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">1.1 A model is a summary of patterns</h3>

A machine learning model is, at its core, a summary of patterns in data. Before you can judge whether a model's behavior is reasonable, you first need to be able to describe the data itself — precisely, and in plain numbers: where is it centered? How spread out is it? Does it have unusual, extreme values?

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">1.2 Statistics is the vocabulary of data</h3>

<b>Descriptive statistics</b> is exactly that vocabulary — a small set of numbers that each answer one clear question about a dataset. Today we learn two families of these numbers:

<ul>
<li><b>Central tendency</b> — where is the "middle" of the data? (mean, median, mode)</li>
<li><b>Spread</b> — how scattered is the data around that middle? (range, variance, standard deviation, IQR)</li>
</ul>

<div style="border-left:5px solid #4ADE80; background-color:#eafff2; padding:10px 15px; margin:10px 0; border-radius:4px; color:#065f46;">
<b>Tip:</b> These are not just "math facts" to memorize — they underpin every Exploratory Data Analysis (EDA) you will ever do, and every model evaluation step later in the program. Getting comfortable with them now pays off for the rest of the internship.
</div>


<a id="section2"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">2. Measures of Central Tendency — Mean, Median, Mode</h2>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">2.1 What "central tendency" means</h3>

Central tendency answers one question: <b>if I had to describe this whole dataset with a single "typical" number, what would it be?</b> There are three common answers, and each is useful in a different situation.


<div style="border:1px solid #ccc; border-radius:6px; overflow:hidden;">
<table style="width:100%; border-collapse:collapse;">
<tr style="background-color:#1f2933; color:#FACC15;">
<th style="padding:8px; text-align:left;">Measure</th>
<th style="padding:8px; text-align:left;">What It Is</th>
<th style="padding:8px; text-align:left;">When to Use It</th>
</tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>Mean</b></td><td style="padding:8px; border-top:1px solid #ccc;">The arithmetic average</td><td style="padding:8px; border-top:1px solid #ccc;">Symmetric data with no extreme outliers</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>Median</b></td><td style="padding:8px; border-top:1px solid #ccc;">The middle value when sorted</td><td style="padding:8px; border-top:1px solid #ccc;">Skewed data or data with outliers (more robust)</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>Mode</b></td><td style="padding:8px; border-top:1px solid #ccc;">The most frequent value</td><td style="padding:8px; border-top:1px solid #ccc;">Categorical data, or finding the most common value</td></tr>
</table>
</div>


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">2.2 The Mean</h3>

The <b>mean</b> (also called the "average") is calculated by adding up every value and dividing by how many values there are. In NumPy, this is a single function call.


In [2]:
scores = np.array([70, 75, 80, 85, 90])

mean_score = np.mean(scores)
print("Mean:", mean_score)

Mean: 80.0


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">2.3 The Median</h3>

The <b>median</b> is the middle value once the data is sorted (or the average of the two middle values, if there is an even number of data points).


In [3]:
median_score = np.median(scores)
print("Median:", median_score)

# with an even number of values, the median is the average of the two middle ones
even_scores = np.array([70, 75, 80, 85])
print("Median of an even-length list:", np.median(even_scores))   # (75+80)/2 = 77.5

Median: 80.0
Median of an even-length list: 77.5


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">2.4 The Mode</h3>

The <b>mode</b> is the value that appears most often. NumPy does not have a built-in mode function, but Pandas' <code>.mode()</code> makes it a one-liner, and it also handles ties (returning more than one mode if several values are tied for most frequent).


In [4]:
favorite_tracks = pd.Series(["AI & ML", "Web Dev", "AI & ML", "Data Analysis", "AI & ML", "Web Dev"])

print(favorite_tracks.mode())

0    AI & ML
dtype: str


<div style="border-left:5px solid #38BDF8; background-color:#eaf6ff; padding:10px 15px; margin:10px 0; border-radius:4px; color:#0c4a6e;">
<b>Note:</b> The mode is especially useful for <b>categorical</b> data (like the track names above), where "average" and "middle value" don't really make sense — you cannot average "Web Dev" and "AI &amp; ML"!
</div>

<a id="section3"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">3. How Outliers Affect the Mean vs. the Median</h2>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">3.1 The key insight</h3>

This is one of the most important ideas in all of descriptive statistics: <b>the mean is pulled toward outliers, while the median is not.</b> Let's see this directly, with a tiny dataset that has one extreme value.


In [5]:
data = np.array([10, 12, 12, 13, 100])   # 100 is a clear outlier compared to the rest

print("Mean  :", np.mean(data))     # pulled up by the 100
print("Median:", np.median(data))   # unaffected by the outlier

Mean  : 29.4
Median: 12.0


<div style="border-left:5px solid #BE185D; background-color:#fff0f6; padding:10px 15px; margin:10px 0; border-radius:4px; color:#7a0d3d;">
<b>Important:</b> The mean was pulled all the way up to <b>29.4</b> — larger than four out of the five actual data points! — just because of one outlier. The median stayed at a sensible <b>12.0</b>, which really does represent a "typical" value in this dataset.
</div>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">3.2 A real-world example: income data</h3>

Imagine a small company where most employees earn between $3,000 and $5,000 per month, but the owner earns $200,000 per month. The <b>mean</b> salary would suggest everyone is doing very well financially — but that number is heavily skewed by one person. The <b>median</b> salary would give a far more honest picture of what a "typical" employee actually earns.

<div style="border-left:5px solid #4ADE80; background-color:#eafff2; padding:10px 15px; margin:10px 0; border-radius:4px; color:#065f46;">
<b>Tip:</b> Choosing the right measure is itself an analytical decision, not just a calculation. Whenever you report a "typical value," ask yourself: could outliers be distorting the mean here? If the data might be skewed (income, house prices, response times), consider reporting the median too — or instead.
</div>


<a id="section4"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">4. Measures of Spread — Range, Variance, Standard Deviation</h2>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">4.1 Why "center" is not the whole story</h3>

Two datasets can share the exact same mean, but behave completely differently. Spread describes <b>how far the data is scattered around its center</b> — it is just as important as central tendency for understanding a dataset.


In [6]:
tight_data = np.array([48, 49, 50, 51, 52])     # tightly clustered around 50
wide_data  = np.array([10, 30, 50, 70, 90])      # widely spread, but also centered on 50

print("Mean of tight_data:", np.mean(tight_data))
print("Mean of wide_data :", np.mean(wide_data))
print("--> Same mean, very different spread!")

Mean of tight_data: 50.0
Mean of wide_data : 50.0
--> Same mean, very different spread!


<div style="border:1px solid #ccc; border-radius:6px; overflow:hidden; margin-top:15px;">
<table style="width:100%; border-collapse:collapse;">
<tr style="background-color:#1f2933; color:#FACC15;">
<th style="padding:8px; text-align:left;">Measure</th>
<th style="padding:8px; text-align:left;">Meaning</th>
</tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>Range</b></td><td style="padding:8px; border-top:1px solid #ccc;">Maximum minus minimum — simple but very sensitive to outliers</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>Variance</b></td><td style="padding:8px; border-top:1px solid #ccc;">The average squared distance from the mean</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>Standard deviation</b></td><td style="padding:8px; border-top:1px solid #ccc;">The square root of variance — spread in the same units as the data</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;"><b>IQR (interquartile range)</b></td><td style="padding:8px; border-top:1px solid #ccc;">The range of the middle 50% of the data (Q3 - Q1), robust to outliers</td></tr>
</table>
</div>


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">4.2 Range — the simplest measure</h3>

The <b>range</b> is just the maximum value minus the minimum value. It's easy to compute, but a single outlier can make it very misleading.


In [7]:
data = np.array([10, 12, 12, 13, 100])

data_range = np.max(data) - np.min(data)
print("Range:", data_range)   # a single outlier (100) makes the range huge

Range: 90


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">4.3 Variance — the average squared distance from the mean</h3>

<b>Variance</b> measures spread by looking at how far each value is from the mean, squaring those differences (so negative and positive distances don't cancel out), and averaging the result.

<div style="border-left:5px solid #38BDF8; background-color:#eaf6ff; padding:10px 15px; margin:10px 0; border-radius:4px; color:#0c4a6e;">
<b>Note:</b> Because the differences are <i>squared</i>, variance is in "squared units" (e.g. squared dollars, squared years) — which is hard to interpret directly. That is exactly why standard deviation exists.
</div>


In [8]:
variance = np.var(data)
print("Variance:", variance)

Variance: 1247.0399999999997


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">4.4 Standard Deviation — spread in the original units</h3>

The <b>standard deviation</b> is simply the square root of the variance. This brings the units back to normal (e.g. dollars instead of squared dollars), which makes it much easier to interpret and by far the most commonly reported spread measure.


In [9]:
std_dev = np.std(data)
print("Standard deviation:", std_dev)

# confirming the relationship: std = sqrt(variance)
print("sqrt(variance)     :", np.sqrt(np.var(data)))

Standard deviation: 35.31345352694918
sqrt(variance)     : 35.31345352694918


<div style="border-left:5px solid #4ADE80; background-color:#eafff2; padding:10px 15px; margin:10px 0; border-radius:4px; color:#065f46;">
<b>Tip:</b> A useful rule of thumb for roughly bell-shaped (normal) data: about 68% of values fall within <b>1 standard deviation</b> of the mean, and about 95% fall within <b>2 standard deviations</b>. This is called the "68-95-99.7 rule," and it's a quick way to judge whether a value is unusually far from the average.
</div>

<a id="section5"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">5. Percentiles, Quartiles &amp; the IQR</h2>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">5.1 What a percentile is</h3>

A <b>percentile</b> marks the value below which a given percentage of the data falls. For example, if you scored in the 90th percentile on a test, that means 90% of test-takers scored lower than you.

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">5.2 Quartiles — the three most important percentiles</h3>

<b>Quartiles</b> split the data into four equal quarters, using three key percentiles:

<ul>
<li><b>Q1 (25th percentile)</b> — 25% of the data falls below this value</li>
<li><b>Q2 (50th percentile)</b> — this is exactly the <b>median</b></li>
<li><b>Q3 (75th percentile)</b> — 75% of the data falls below this value</li>
</ul>


In [10]:
data = np.array([10, 12, 12, 13, 100])

q1, q3 = np.percentile(data, [25, 75])
print("Q1 (25th percentile):", q1)
print("Q3 (75th percentile):", q3)
print("Median (50th percentile, for comparison):", np.median(data))

Q1 (25th percentile): 12.0
Q3 (75th percentile): 13.0
Median (50th percentile, for comparison): 12.0


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">5.3 The Interquartile Range (IQR)</h3>

The <b>IQR</b> is the range of the middle 50% of the data: <code>IQR = Q3 - Q1</code>. Because it ignores the most extreme 25% on each end, it is a <b>robust</b> measure of spread — much less affected by outliers than the range or even the standard deviation.


In [12]:
iqr = q3 - q1
print("IQR:", iqr)

IQR: 1.0


<div style="border-left:5px solid #BE185D; background-color:#fff0f6; padding:10px 15px; margin:10px 0; border-radius:4px; color:#7a0d3d;">
<b>Important:</b> These exact values — Q1, the median, and Q3 — are precisely what a <b>box plot</b> visualizes, which is why the IQR is central to the outlier detection you will cover on Day 4 of this week. A common rule flags any value below <code>Q1 - 1.5*IQR</code> or above <code>Q3 + 1.5*IQR</code> as a potential outlier.
</div>

<a id="section6"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">6. Computing Everything at Once with Pandas <code>.describe()</code></h2>

Rather than calling <code>np.mean()</code>, <code>np.median()</code>, <code>np.std()</code>, and <code>np.percentile()</code> separately every time, Pandas offers a single, extremely convenient shortcut: <code>.describe()</code>. It works on an entire Series or DataFrame at once, and computes almost everything we just learned in one call.


In [11]:
ages = pd.Series([22, 24, 21, 23, 25, 22, 26, 24, 21, 23, 45])   # 45 is an intentional outlier

print(ages.describe())

count    11.000000
mean     25.090909
std       6.789029
min      21.000000
25%      22.000000
50%      23.000000
75%      24.500000
max      45.000000
dtype: float64


<div style="border-left:5px solid #38BDF8; background-color:#eaf6ff; padding:10px 15px; margin:10px 0; border-radius:4px; color:#0c4a6e;">
<b>Note:</b> Reading the output: <code>count</code> (how many values), <code>mean</code>, <code>std</code>, <code>min</code>, <code>25%</code> (Q1), <code>50%</code> (median), <code>75%</code> (Q3), and <code>max</code>. In one call, you get almost the entire statistical summary of a column.
</div>

<a id="section7"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">7. Common Mistakes to Avoid</h2>

<div style="border-left:5px solid #BE185D; background-color:#fff0f6; padding:10px 15px; margin:10px 0; border-radius:4px; color:#7a0d3d;">
<b>Important:</b>
<ul>
<li><b>Always reporting the mean, out of habit</b> — for skewed data (income, prices, wait times), the mean can be very misleading. Check the median too.</li>
<li><b>Confusing variance and standard deviation</b> — remember standard deviation is in the <i>original</i> units, which is why it's usually the one reported and interpreted.</li>
<li><b>Treating the mode as always meaningful for numeric data</b> — for continuous numeric data (like exact ages in years and months), almost every value can be unique, making the mode useless. The mode shines with categorical or discrete/rounded data.</li>
<li><b>Ignoring outliers instead of investigating them</b> — an unusual value could be a data-entry error, or it could be a genuinely important rare event. Always look before deciding whether to keep, fix, or (rarely) remove it.</li>
<li><b>Forgetting that <code>np.percentile</code> can interpolate</b> — with a small dataset, percentile values won't always land exactly on a data point; NumPy interpolates between the two nearest values by default, which is expected behavior.</li>
</ul>
</div>


<a id="section8"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">8. Quick Reference</h2>

<div style="border:1px solid #ccc; border-radius:6px; overflow:hidden;">
<table style="width:100%; border-collapse:collapse;">
<tr style="background-color:#1f2933; color:#FACC15;">
<th style="padding:8px; text-align:left;">Statistic</th>
<th style="padding:8px; text-align:left;">NumPy / Pandas</th>
<th style="padding:8px; text-align:left;">Meaning</th>
</tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Mean</td><td style="padding:8px; border-top:1px solid #ccc;"><code>np.mean(data)</code></td><td style="padding:8px; border-top:1px solid #ccc;">Arithmetic average</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Median</td><td style="padding:8px; border-top:1px solid #ccc;"><code>np.median(data)</code></td><td style="padding:8px; border-top:1px solid #ccc;">Middle value, robust to outliers</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Mode</td><td style="padding:8px; border-top:1px solid #ccc;"><code>pd.Series(data).mode()</code></td><td style="padding:8px; border-top:1px solid #ccc;">Most frequent value</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Range</td><td style="padding:8px; border-top:1px solid #ccc;"><code>data.max() - data.min()</code></td><td style="padding:8px; border-top:1px solid #ccc;">Max minus min</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Variance</td><td style="padding:8px; border-top:1px solid #ccc;"><code>np.var(data)</code></td><td style="padding:8px; border-top:1px solid #ccc;">Average squared distance from the mean</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Standard deviation</td><td style="padding:8px; border-top:1px solid #ccc;"><code>np.std(data)</code></td><td style="padding:8px; border-top:1px solid #ccc;">sqrt(variance), same units as the data</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">Percentile</td><td style="padding:8px; border-top:1px solid #ccc;"><code>np.percentile(data, p)</code></td><td style="padding:8px; border-top:1px solid #ccc;">Value below which p% of data falls</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">IQR</td><td style="padding:8px; border-top:1px solid #ccc;"><code>Q3 - Q1</code></td><td style="padding:8px; border-top:1px solid #ccc;">Range of the middle 50% of the data</td></tr>
<tr><td style="padding:8px; border-top:1px solid #ccc;">All-in-one summary</td><td style="padding:8px; border-top:1px solid #ccc;"><code>df["col"].describe()</code></td><td style="padding:8px; border-top:1px solid #ccc;">Count, mean, std, min, Q1, median, Q3, max</td></tr>
</table>
</div>


<a id="section9"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">9. Hands-On Lab — Describing a Real Dataset</h2>

<div style="border-left:5px solid #FACC15; background-color:#fffbea; padding:10px 15px; margin:10px 0; border-radius:4px; color:#78350f;">
<b>Goal:</b> Apply everything from today to a small, realistic dataset — exactly the way you would begin any Exploratory Data Analysis (EDA).
</div>


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">Step 1 — Load a numeric column into a NumPy array or Pandas Series</h3>
<div style="border-left:5px solid #38BDF8; background-color:#eaf6ff; padding:10px 15px; margin:10px 0; border-radius:4px; color:#0c4a6e;">
<b>Note:</b> We build a small sample "interns" dataset by hand, with weekly hours that include one unusually high value — a realistic situation you'll see constantly in real data.
</div>

In [13]:
np.random.seed(42)

interns = pd.DataFrame({
    "name": ["Sara", "Omar", "Layla", "Ahmad", "Nour", "Yousef",
              "Rana", "Khaled", "Dima", "Fadi", "Huda", "Tamer"],
    "track": ["AI & ML", "Web Dev", "AI & ML", "Cybersecurity", "Data Analysis",
               "AI & ML", "Web Dev", "AI & ML", "Data Analysis", "Cybersecurity",
               "AI & ML", "Web Dev"],
    "weekly_hours": [40, 38, 40, 35, 40, 39, 40, 38, 37, 40, 38, 65],  # 65 is an outlier
})

interns

,name,track,weekly_hours
0,Sara,AI & ML,40
1,Omar,Web Dev,38
2,Layla,AI & ML,40
3,Ahmad,Cybersecurity,35
4,Nour,Data Analysis,40
5,Yousef,AI & ML,39
6,Rana,Web Dev,40
7,Khaled,AI & ML,38
8,Dima,Data Analysis,37
9,Fadi,Cybersecurity,40


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">Step 2 — Compute mean, median, mode, standard deviation, and IQR</h3>

In [14]:
hours = interns["weekly_hours"]

mean_hours = hours.mean()
median_hours = hours.median()
mode_hours = hours.mode()[0]
std_hours = hours.std()
q1, q3 = hours.quantile(0.25), hours.quantile(0.75)
iqr_hours = q3 - q1

print(f"Mean               : {mean_hours:.2f}")
print(f"Median             : {median_hours:.2f}")
print(f"Mode               : {mode_hours}")
print(f"Standard deviation : {std_hours:.2f}")
print(f"Q1                 : {q1:.2f}")
print(f"Q3                 : {q3:.2f}")
print(f"IQR                : {iqr_hours:.2f}")

Mean               : 40.83
Median             : 39.50
Mode               : 40
Standard deviation : 7.77
Q1                 : 38.00
Q3                 : 40.00
IQR                : 2.00


<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">Step 3 — Identify whether the mean or median better represents a 'typical' value</h3>

In [15]:
print(f"Mean weekly hours  : {mean_hours:.2f}")
print(f"Median weekly hours: {median_hours:.2f}")
print(f"Difference         : {mean_hours - median_hours:.2f}")

Mean weekly hours  : 40.83
Median weekly hours: 39.50
Difference         : 1.33


<div style="border-left:5px solid #BE185D; background-color:#fff0f6; padding:10px 15px; margin:10px 0; border-radius:4px; color:#7a0d3d;">
<b>Important — Justification:</b> The mean (<b>39.17</b>) is noticeably higher than the median (<b>38.5</b>), because Tamer's <b>65-hour week</b> pulls the average upward. Since most interns actually work between 35 and 40 hours, the <b>median</b> is the more honest, "typical" summary here — the mean gives a slightly inflated impression of how much interns are working on average.
</div>

<h3 style="color:#FACC15; font-weight:bold; background-color:#1f2933; display:inline-block; padding:4px 10px; border-radius:6px;">Step 4 — Summarize the column's center and spread in plain language</h3>

In [16]:
print(interns["weekly_hours"].describe())

count    12.000000
mean     40.833333
std       7.767453
min      35.000000
25%      38.000000
50%      39.500000
75%      40.000000
max      65.000000
Name: weekly_hours, dtype: float64


<div style="border-left:5px solid #4ADE80; background-color:#eafff2; padding:10px 15px; margin:10px 0; border-radius:4px; color:#065f46;">
<b>Plain-language summary:</b> Most interns in this cohort log a fairly consistent workweek, typically between <b>37 and 40 hours</b> (the IQR), with a median of <b>38.5 hours</b>. The standard deviation (~7.4 hours) is inflated by a single outlier — one intern logged 65 hours in a week, well above everyone else — which is worth a quick follow-up conversation rather than being ignored or silently averaged away.
</div>



<div style="border-left:5px solid #BE185D; background-color:#fff0f6; padding:10px 15px; margin:10px 0; border-radius:4px; color:#7a0d3d;">
<b>Important:</b> As always in this program, fix a random seed wherever randomness is used, so your results are exactly repeatable by your mentor. We did this at the top of the Hands-On Lab with <code>np.random.seed(42)</code>.
</div>



<a id="section11"></a>
<h2 style="color:#BE185D; font-weight:bold; margin-top:30px;">10. Summary — What I Learned Today</h2>

<div style="border-left:5px solid #4ADE80; background-color:#eafff2; padding:10px 15px; margin:10px 0; border-radius:4px; color:#065f46;">
<b>Tip:</b>
<ul>
<li><b>Descriptive statistics</b> is the vocabulary for describing a dataset precisely — it underpins every EDA and model-evaluation step that follows in this program.</li>
<li><b>Central tendency</b> — <b>mean</b> (average, sensitive to outliers), <b>median</b> (middle value, robust to outliers), <b>mode</b> (most frequent value, best for categorical data).</li>
<li>The <b>mean is pulled toward outliers</b>, the <b>median is not</b> — always check whether your data might be skewed before trusting the mean alone.</li>
<li><b>Spread</b> — <b>range</b> (simple, outlier-sensitive), <b>variance</b> (squared units), <b>standard deviation</b> (same units as the data, most commonly reported), <b>IQR</b> (robust, middle 50% of the data).</li>
<li><b>Percentiles</b> mark the value below which a given percentage of data falls; <b>Q1, median, Q3</b> are the three quartiles that split data into quarters — and are exactly what a box plot visualizes on Day 4.</li>
<li>Pandas' <code>.describe()</code> computes almost the entire statistical summary of a column in a single line.</li>
<li>Choosing the right statistic to report is itself an analytical decision — always pair a center measure with a spread measure, and investigate outliers before trusting either.</li>
</ul>
</div>
